src/components/CodeView/utils.spec.tsx

```diff
@@ -16,4 +19,138 @@ describe(__filename, () => {
       expect(getLines(content)).toEqual(lines);
     });
   });
+
+  describe('map functions', () => {
+    const createChild = ({
+      children,
+      properties,
+      tagName,
+      type = '',
+      value = '',
+    }: Partial<RefractorNode>): RefractorNode => {
+      return {
+        children,
+        properties,
+        tagName,
+        type,
+        value,
+      };
+    };
+
+    describe('mapChild', () => {
+      it('creates a React element from a child with a tagName and a className', () => {
```

Comment:
In addition to `className`, I'd like to see coverage of arbitrary properties. This test should do it but the TS types need adjusting somehow.

```diff
diff --git a/src/components/CodeView/utils.spec.tsx b/src/components/CodeView/utils.spec.tsx
index 9a3a839..1023a8e 100644
--- a/src/components/CodeView/utils.spec.tsx
+++ b/src/components/CodeView/utils.spec.tsx
@@ -52,6 +52,17 @@ describe(__filename, () => {
         expect(element).toHaveProperty('props.className', className);
       });
 
+      it('passes all child properties to the element', () => {
+        const target = '_blank';
+        const title = 'link title';
+        const child = createChild({ tagName: 'a', properties: { target, title } });
+
+        const element = mapChild(child, 0, 0);
+
+        expect(element).toHaveProperty('props.target', target);
+        expect(element).toHaveProperty('props.title', title);
+      });
+
       it('creates a React element with strictly positive depth and index', () => {
         const depth = 2;
         const i = 4;
```

Reply:
Added, thanks!